### ISSUE TITLE
`Enter` just takes me back to the last buffer

### ISSUE THREAD

**Issue by gilbox** [FichteFoll/FileHistory#36]
If I currently have file `A.js` open, and then open the *FileHistory* list I will see files `A.js`, `B.js`, `C.js`, etc. Now let's say I press `down` two times. I will see in the quick preview `C.js` (as expected). However, if I now press `return` it will take me back to `A.js`. Alternatively, if I press `esc` I will remain in `C.js`. So, currently I always try to press `esc` but it's rather unintuitive and I often accidentally press `return`

to summarize...

doesn't work:
`recent_active_files` -> `down` -> `down` -> `return`

works:
`recent_active_files` -> `down` -> `down` -> `esc`

**Comment by FichteFoll** [comment]
Can read through #34 and see whether you run into the same issue? Are there any exceptions in the console after any of the two actions (return/esc)? What is your ST version?

**Comment by gilbox** [comment]
No, I don't think it's the same as #34. My "current" buffer remains open, but it does switch to whatever buffer was selected when I press `esc`.

There are no exceptions in the console in either case. I have ST3, build 3103, although when I posted this issue I was on an earlier version. Both exhibit the same behavior.

**Comment by FichteFoll** [comment]
I have just pushed [a new version](https://github.com/FichteFoll/FileHistory/releases/tag/v1.8.0) that may have fixed you problem. Please let me know whether the issue is fixed after you updated.

**Comment by gilbox** [comment]
Can I upgrade via package control? Doesn't seem to be working.

**Comment by FichteFoll** [comment]
You may have to wait up to an hour until the update gets picked up.

**Comment by gilbox** [comment]
I managed to upgrade to `1.8.0` but the bug persists.

**Comment by FichteFoll** [comment]
Please paste your settings file and the contents of the console after you enabled the debug setting and replicated the issue.

**Comment by gilbox** [comment]
settings file was empty so I added this:
```
{
    "debug": true
}
```

the output was:

```
[FileHistory] Reloading the settings file "FileHistory.sublime-settings".
[FileHistory] Setting "debug" = True
[FileHistory] Setting "global_max_entries" = 100
[FileHistory] Setting "project_max_entries" = 50
[FileHistory] Setting "use_saved_position" = True
[FileHistory] Setting "new_tab_position" = 'next'
[FileHistory] Setting "reopen_file_in_current_group" = False
[FileHistory] Setting "remove_non_existent_files_on_preview" = False
[FileHistory] Setting "cleanup_on_startup" = True
[FileHistory] Setting "delete_all_on_startup" = False
[FileHistory] Setting "history_file" = 'User/FileHistory.json'
[FileHistory] Setting "monospace_font" = False
[FileHistory] Setting "timestamp_show" = True
[FileHistory] Setting "timestamp_format" = '%Y-%m-%d @ %H:%M:%S'
[FileHistory] Setting "timestamp_mode" = 'history_access'
[FileHistory] Setting "timestamp_relative" = True
[FileHistory] Setting "prettify_history" = False
[FileHistory] Setting "path_exclude_patterns" = []
[FileHistory] Setting "path_reinclude_patterns" = []
[FileHistory] Setting "max_backup_count" = 3
[FileHistory] Setting "show_file_preview" = True
```

replicating the issue provided no additional output

**Comment by FichteFoll** [comment]
That is incredibly odd. This is what my console looks like when opening the File History quick panel, pressing the down key twice and then opening the selected entry with enter:

```
[FileHistory] Opening preview for 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\Default\Preferences.sublime-settings'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\Default\Preferences.sublime-settings'
[FileHistory] Opening preview for 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\User\Preferences.sublime-settings'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\Default\Preferences.sublime-settings'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\User\Preferences.sublime-settings'
[FileHistory] Opening preview for 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\User\Preferences.sublime-settings'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py'
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py'
[FileHistory] Closing file: C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py
[FileHistory] Detected possibly transient view with index = -1: 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py'
SublimeLinter: annotations activated: <builtin> 
[FileHistory] Opened file in group 1, index 3 (based on saved group 1, index 3): C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py
[FileHistory] Opening preview for 'C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py'
[FileHistory] Adding opened file to project "b0dd96922453807242be2caeb68b3206" with group 1 and index 3: C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py
[FileHistory] Adding opened file to project "global" with group 1 and index 3: C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\PackageDev\scope_data\__init__.py
[FileHistory] Saving the history to file C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\User\FileHistory.json
[FileHistory] Backing up the history file for 20160407
[FileHistory] Discarding old backup C:\Users\Fichte\AppData\Roaming\Sublime Text 3\Packages\User\FileHistory_20160404.json
```

I'm completely at loss why not even the first line was printed. Could you record a video of you reproducing the bug with the console open? Make sure you have the debug setting enabled and entered `sublime.log_commands(True)` in the console before.

**Comment by gilbox** [comment]
Sorry, I may not have done it the way you expected the first time. Here is a screencap. Note that the first two times I press the `Esc` key and it switches to the tab I want. The other times I am pressing `return`.

![fh](https://cloud.githubusercontent.com/assets/2136203/14361528/1ff594c6-fcb0-11e5-9bcd-258aec53b619.gif)

**Comment by FichteFoll** [comment]
Ah, I know now.

`recent_active_files` is not a command provided by this package, but rather [RecentActiveFiles](https://packagecontrol.io/packages/RecentActiveFiles). The popup also looks different to the one previewed in this package's readme.

Simply said: You have two packages for opening previously accessed files and one (not this) is behaving badly. Just remove RecentActiveFiles and you should be good. The problems you seem to be running into are documented in this issue: https://github.com/jugyo/SublimeRecentActiveFiles/issues/2

**Comment by gilbox** [comment]
doh sorry about that
